docs/source/blueprints/BP000.rst

```diff
@@ -0,0 +1,307 @@
+:Number:BP000
+:Title: Blueprint specification and requirements
+:Author: Beraldo Leal <bleal@redhat.com>
+:Reviewers: 
+:Created: 29-Sep-2020
+:Type: Meta Blueprint
+:Status: Draft
+
+.. contents:: Table of Contents
+
+TL;DR
+*****
+
+Having better records of architectural decisions in our repository is a good
+way to socialize our technical decisions, improve the code review, avoid
+unnecessary code, and load balance the workload among other developers. We are
+already using the “blueprint” documents for a while, and still learning how to
+write them. Although we have a basic idea of what is a blueprint, some may not
+understand the concept because we are missing one meta-document that describes
+the blueprint's basic notions. This document is a blueprint to specify how we
+should write blueprints from now on.
+
+Motivation
+**********
+
+Depending on the project size, having very well defined and structured
+documents about the architecture decisions seems overkill at first glance, but
+could help any size projects, including Avocado, save time, make better
+decisions and improve the way we socialize those decisions.
+
+Today we have inside the Avocado project the good practice to submit an RFC to
+our mailing list or using a GitHub issue when we have a new idea. RFCs are a
+widespread way to disclose the architecture decisions, but they are just one
+part of the process. During the RFC phase, we argue in favor of a proposal, and
+we are mostly concerned about collecting feedback. After this phase, we could
+go one step forward and consolidate the discussion in Blueprints (sometimes
+called ADR - Architecture Decision Records). This could be the next step so we
+could better socialize our decisions for future readers. A very well defined
+and structured document has some advantages over an RFC, but it is not intended
+to replace it.
```

Comment:
I guess, that we miss some sections on this BP yet. I was thinking about this. "When to write a BP" is a missing section here.

IMO, we could have only a BP if the author is familiar with the BP structure and has confident that topic is a BP suitable. But sometimes, new members don't know about BP, or are not 100% sure if it is a good idea. In this case, makes sense to send a RFC first with a more informal discussion as a preparation for a BP. Makes sense?

Reply:
Yes, for sure a new user would ask themselves what is the difference between an RFC and a blueprint but at least the explanation here doesn't seem full clear to me. We can simplify and go straight to the point with a sentence or two saying that we understand a blueprint as more formal document, containing and RFC within itself and additional points (...). Then we could mention that a user could start with the less formal RFC within the document instead of the general statements that RFC remain irreplaceable which made me wonder "in what aspects?".

Reply:
Yes, the terms are confusing.... In this document, when I talk about RFC, I'm talking about the "open format email" that we sent to the mailing list (with RFC in the subject).

The sentence is there just to make clear that it was not my intetion to abolish RFC (as we make it today). And members could still use the mailing list to start a discussion, by sending a RFC. 

But in fact, RFC could be just any "request for comments": an email starting a discussion with no RFC in the subject; or even a PR with a blueprint in draft mode.

It is just a difference about the approach, instead of trying to make our RFC more formal, I'm just introducing a new model (blueprints) that is the formal "version" of our RFCs.

So, what I was trying to say was: "By introducing this blueprint model we don't need to abolish the RFC as we use today, users can still send RFCs"

But yes, we need to make this clear in this document.